middleware.ml

```diff
@@ -123,21 +128,29 @@ let cookie_value cookie =
   | _ -> Error (`Msg "Bad cookie")
 
 let user_from_auth_cookie cookie users =
-  match cookie_value cookie with
-  | Ok cookie_value -> (
-      match User_model.find_user_by_key cookie_value users with
-      | Some user -> Ok user
-      | None -> Error (`Msg "User not found"))
-  | Error (`Msg s) ->
-      Logs.err (fun m -> m "Error: %s" s);
-      Error (`Msg s)
+  match User_model.find_user_by_key cookie users with
+  | Some user -> Ok user
+  | None -> Error (`Msg "User not found")
 
-let user_of_cookie users now reqd =
+let get_cookie_from_request reqd =
   match has_cookie "molly_session" reqd with
   | Some auth_cookie -> (
+      match cookie_value auth_cookie with
+      | Ok cookie_value -> Ok cookie_value
+      | Error (`Msg s) ->
+          Logs.err (fun m -> m "Error: %s" s);
+          Error (`Msg s))
+  | None ->
+      Logs.err (fun m ->
+          m "auth-middleware: No molly-session in cookie header.");
+      Error (`Msg "User not found")
+
+let user_of_cookie users now reqd =
+  match get_cookie_from_request reqd with
+  | Ok auth_cookie -> (
       match user_from_auth_cookie auth_cookie users with
       | Ok user -> (
-          match User_model.user_auth_cookie_from_user user with
+          match User_model.user_auth_cookie_from_user auth_cookie user with
```

Comment:
I'm pretty unhappy with the function name: it takes an `auth_cookie` and an `user` and returns `user_auth_cookie`? Needs to be clarified (once I reach User_model, this may become clearer to me).

Reply:
it is ok, since what the function does is to retrieve the right cookie (whose cookie_value is the same as given).